internal/receiver/discoveryreceiver/statement_evaluator.go

```diff
@@ -62,14 +63,22 @@ func newStatementEvaluator(logger *zap.Logger, id component.ID, config *Config,
 			config:        config,
 			correlations:  correlations,
 			alreadyLogged: &sync.Map{},
-			// TODO: provide more capable env w/ resource and log record attributes
-			exprEnv: func(pattern string) map[string]any {
-				return map[string]any{"msg": pattern}
-			},
-			id: id,
+			id:            id,
 		},
 		encoder: encoder,
 	}
+
+	se.evaluator.exprEnv = func(pattern string) map[string]any {
```

Comment:
Looks like this is defining evaluator behavior in a foreign constructor using an anonymous function. Can we encapsulate the construction and behavior of `evaluator` in evaluator.go?